lib/salus/scanners/osv/gradle_osv.rb

```diff
@@ -0,0 +1,157 @@
+require 'salus/scanners/osv/base'
+
+module Salus::Scanners::OSV
+  class GradleOSV < Base
+    class SemVersion < Gem::Version; end
+
+    EMPTY_STRING = "".freeze
+    DEFAULT_SOURCE = "https://osv.dev/list".freeze
+    DEFAULT_SEVERITY = "MODERATE".freeze
+    GRADLE_OSV_ADVISORY_URL = "https://osv-vulnerabilities.storage.googleapis.com"\
+        "/Maven/all.zip".freeze
+
+    def should_run?
+      @repository.build_gradle_present?
+    end
+
+    def run
+      # Find dependencies
+      dependencies = find_dependencies
+      if dependencies.empty?
+        err_msg = "GradleOSV: Failed to parse any dependencies from the project."
+        report_stderr(err_msg)
+        report_error(err_msg)
+        return
+      end
+
+      # Fetch vulnerabilities
+      @osv_vulnerabilities ||= fetch_vulnerabilities(GRADLE_OSV_ADVISORY_URL)
+      if @osv_vulnerabilities.nil?
+        err_msg = "GradleOSV: No vulnerabilities found to compare."
+        report_stderr(err_msg)
+        report_error(err_msg)
+        return
+      end
+
+      # Match and Report scanner status
+      vulnerabilities_found = match_vulnerable_dependencies(dependencies)
+      results = group_vulnerable_dependencies(vulnerabilities_found)
+      return report_success if results.empty?
+
+      report_failure
+      log(JSON.pretty_generate(results))
+    end
+
+    private
+
+    # Match if dependency version found is in the range of
+    # vulnerable dependency found.
+    def version_matching(version, introduced, fixed)
+      vulnerable_flag = false
+      if introduced.present? && fixed.present?
+        if SemVersion.new(version) >= SemVersion.new(introduced) &&
+            SemVersion.new(version) < SemVersion.new(fixed)
+          vulnerable_flag = true
+        end
+      elsif introduced.present?
+        vulnerable_flag = true if SemVersion.new(version) >= SemVersion.new(introduced)
+      end
+
+      vulnerable_flag
+    end
+
+    # Compare vulnerabilities found with dependencies found
+    # and return vulnerable dependencies
+    def match_vulnerable_dependencies(dependencies)
+      results = []
+      dependencies.each do |dependency|
+        lib = "#{dependency['group_id']}:#{dependency['artifact_id']}"
+
+        if dependency['version'].present?
+          version = dependency['version']
+          # Cleanup version string.
+          version = version.delete("^0-9.").gsub(/\.+$/, "")
```

Comment:
a short comment that describes why this `delete ... gsub...` is needed would be helpful.